### PR TITLE
Update CI/CD GHA scripts and nginx config.

### PR DESCRIPTION
**Pull request by pwei1018**
*Issue #:* https://github.com/bcgov/entity/issues/4997

*Description of changes:*
1. Update CI/CD github action scripts and update github secrets;
2. Update nginx config;
3. Move build image namespace from gl2uos-tools to mpgxod-tools;
3. Using nginx docker build instead of caddy build;
4. Modify ui configuration configmap to using keycloak config inside the project;

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: .github/workflows/cd.yml
Comment (severinbeauvais) [inline .github/workflows/cd.yml:10]: Dev only?

File: .github/workflows/cd.yml
Comment (severinbeauvais) [inline .github/workflows/cd.yml:29]: Path?

File: .github/workflows/ci.yml
Comment (severinbeauvais) [inline .github/workflows/ci.yml:1]: FWIW, the repo is named Business **Filings** UI. If possible we should use the same labels everywhere.

File: nginx.conf
Comment (severinbeauvais) [inline nginx.conf:31]: Nice to have compression enabled! This should help with performance.
Can this be used in other 2 UIs as well?

Also, do you know how to set up caching for static assets? This could be a new ticket. (See Chrome Lighthouse Performance audit for opportunities.)

File: nginx.conf
Comment (severinbeauvais) [inline nginx.conf:38]: Can this be used in other 2 UIs as well?

**Review by severinbeauvais — APPROVED** [review]
LGTM.

**Comment by severinbeauvais** [comment]
If any OS config has changed, will you update the JSON/param files in the GH project accordingly (so we can rebuild via script if needed).

**Comment by severinbeauvais** [comment]
> 3. Move build image namespace from gl2uos-tools to mpgxod-tools;
Can we store this config (or an explanation for it) somewhere? (It uses the secrets, right?)

> 5. Modify ui configuration configmap to using keycloak config inside the project;
I don't see this change anywhere.

**Review by rarmitag — APPROVED** [review]
good suggestions Severin

File: .github/workflows/cd.yml
Comment (pwei1018) [reply .github/workflows/cd.yml:10]: no, it can be deploy to test use this github event.

File: .github/workflows/ci.yml
Comment (pwei1018) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/cd.yml
Comment (pwei1018) [reply .github/workflows/cd.yml:29]: The composite action  is a new thing.  So, I just put it into my own repo. I have other change plan of this action waiting for the new features release from github.

File: nginx.conf
Comment (pwei1018) [reply nginx.conf:31]: Done. 

You can create a ticket to address the caching things.

File: nginx.conf
Comment (pwei1018) [reply nginx.conf:38]: Done.

**Comment by pwei1018** [comment]
> > 1. Move build image namespace from gl2uos-tools to mpgxod-tools;
> >    Can we store this config (or an explanation for it) somewhere? (It uses the secrets, right?)
> 
> > 1. Modify ui configuration configmap to using keycloak config inside the project;
> >    I don't see this change anywhere.
from https://dev.bcregistry.ca/business/**auth**/config/kc/keycloak.json to https://dev.bcregistry.ca/business/config/kc/keycloak.json in the configmap (https://console.pathfinder.gov.bc.ca:8443/console/project/1rdehl-dev/browse/config-maps/business-filings-dev-configuration)

File: .github/workflows/cd.yml
Comment (severinbeauvais) [reply .github/workflows/cd.yml:10]: Ah, I see. Or any environment I guess?

File: .github/workflows/cd.yml
Comment (severinbeauvais) [reply .github/workflows/cd.yml:29]: Very cool.
I'd feel more comfortable if this used a bcgov repo even now, but I'll leave it up to you.

File: k8s/filings-ui.dc-srvc-rt-cmap-kcmap.param
Comment (severinbeauvais) [inline k8s/filings-ui.dc-srvc-rt-cmap-kcmap.param:24]: This URL should point to the Businesses home page.
Your change will load the Filings UI, which will then load Auth Web (2 steps).
Please revert.

(Yes, we should possibly rename this and/or clean up our usage of this URL, but that's outside the scope of this ticket.)

File: k8s/filings-ui.dc-srvc-rt-cmap-kcmap.param
Comment (severinbeauvais) [inline k8s/filings-ui.dc-srvc-rt-cmap-kcmap.param:31]: Good change. We were sharing the auth keycloack.json file but now we will use our own :)

File: nginx.conf
Comment (severinbeauvais) [reply nginx.conf:31]: Thanks!
I've created a new ticket (https://github.com/bcgov/entity/issues/5037).

File: nginx.conf
Comment (severinbeauvais) [reply nginx.conf:38]: Awesome, thanks!

File: k8s/filings-ui.dc-srvc-rt-cmap-kcmap.param
Comment (pwei1018) [reply k8s/filings-ui.dc-srvc-rt-cmap-kcmap.param]: ok.

**Pull request merged**
